test/evp_test.c

```diff
@@ -3540,6 +3546,18 @@ static int kdf_test_run(EVP_TEST *t)
         t->err = "KDF_CTRL_ERROR";
         goto err;
     }
+
+    if (sk_OPENSSL_STRING_num(expected->final_controls) > 0) {
```

Comment:
Why do you need a 'final_controls'?
The order in the file should determine the order they are set as params.

Reply:
If it is worth it, I would add it into evp_kdf_test.c instead.

Reply:
After reviewing again, I think it is not needed.

Removed.